app/javascript/src/components/Dashboard/Notes/NotesCard.jsx

```diff
@@ -1,31 +1,81 @@
-        <h3>{note.title}</h3>
-        <MenuVertical size={24} color={"gray"} />
-      </div>
-      <div className="text-gray pt-2 px-4">
-        <p className="text-sm pr-1">{note.content}</p>
-      </div>
-      <div className="w-full px-4 pt-2">
-        <div className="border-b neeto-ui-gray-200 w-full"></div>
-      </div>
-      <div className="w-full px-4 flex pt-4 mb-4 justify-between">
-        <Tag className="bg-gray-100 text-gray-500" label="Getting Started" />
-        <div className="gap-x-2 flex">
-          <Clock className="my-auto" size={10} />
-          <span className="my-auto text-sm">Created 2 hours ago</span>
-          <Avatar user={{ name: "Test User", imageUrl }} size="medium" />
+    <>
+      <div className="border mt-5 neeto-ui-border-gray-300 neeto-ui-shadow-s">
+        <div className="pt-4 px-4 flex w-full justify-between">
+          <h3>{note.title}</h3>
+          <Dropdown
+            icon={() => <MenuVertical size={20} color={"gray"} />}
+            position="bottom-start"
+            buttonStyle="text"
+          >
+            <li>Edit</li>
+            <li onClick={() => setShowModal(true)}>Delete</li>
+          </Dropdown>
+        </div>
+        <div className="text-gray pt-2 px-4">
+          <p className="text-sm pr-1">{note.content}</p>
+        </div>
+        <div className="w-full px-4 pt-2">
+          <div className="border-b neeto-ui-gray-200 w-full"></div>
+        </div>
+        <div className="w-full px-4 flex pt-4 mb-4 justify-between">
+          <Tag className="bg-gray-100 text-gray-500" label={note.tag} />
+          <div className="gap-x-2 flex">
+            <Clock className="my-auto" size={10} />
+            <Tooltip
+              content={<span>Wednesday, 10:30AM</span>}
+              placement="bottom-start"
+            >
+              <span className="my-auto text-sm">Created 2 hours ago</span>
+            </Tooltip>
+            <Avatar user={{ name: "Test User", imageUrl }} size="medium" />
+          </div>
         </div>
       </div>
-    </div>
+      <Modal isOpen={showModal} onClose={() => setShowModal(false)} size={"md"}>
+        <Modal.Header>
+          <Typography style="h2">Delete Note</Typography>
+        </Modal.Header>
+        <Modal.Body>
+          <Typography style="body2" lineHeight="normal">
+            Are you sure you want to delete the note? This action cannot be
+            undone.
+          </Typography>
+        </Modal.Body>
+        <Modal.Footer className="space-x-2">
+          <Button
+            label="Continue"
+            onClick={() => {
+              deleteNote(note.id);
+              setShowModal(false);
+            }}
+            size="large"
+          />
+          <Button
+            style="text"
+            label="Cancel"
+            onClick={() => setShowModal(false)}
+            size="large"
+          />
+        </Modal.Footer>
+      </Modal>
+    </>
```

Comment:
The same reason why I would suggest component splitting here. 
Also, let's keep `Modal` in the parent component and trigger it based on state.